google-cloud-tools-plugin/testSrc/com/google/cloud/tools/intellij/apis/CloudLibraryProjectStateTest.java

```diff
@@ -0,0 +1,242 @@
+                  LIBRARY.clients().get(0).mavenCoordinates();
+              String groupId = mavenCoordinates.groupId();
+              String artifactId = mavenCoordinates.artifactId();
+
+              MavenId managedDependency = new MavenId(groupId, artifactId, "1.0");
+              MavenId unmanagedDependency = new MavenId("my-group", "my-artifact", "1.0");
+              writeDependenciesToPom(
+                  module, ImmutableList.of(managedDependency, unmanagedDependency));
+
+              state.syncManagedProjectLibraries();
+
+              Set<CloudLibrary> libraries = state.getManagedLibraries(module);
+              assertThat_managedLibrariesContainsExactlyOne(libraries);
+            });
+  }
+
+  @Test
+  public void managedLibraries_withOnlyManagedDependencies_isPresent() {
+    when(librariesService.getCloudLibraries())
+        .thenReturn(ImmutableList.of(LIBRARY.toCloudLibrary()));
+
+    ApplicationManager.getApplication()
+        .invokeAndWait(
+            () -> {
+              Module module = createNewModule(id);
+
+              TestCloudLibraryClientMavenCoordinates mavenCoordinates =
+                  LIBRARY.clients().get(0).mavenCoordinates();
+              String groupId = mavenCoordinates.groupId();
+              String artifactId = mavenCoordinates.artifactId();
+
+              MavenId managedDependency = new MavenId(groupId, artifactId, "1.0");
+              writeDependenciesToPom(module, ImmutableList.of(managedDependency));
+
+              state.syncManagedProjectLibraries();
+
+              Set<CloudLibrary> libraries = state.getManagedLibraries(module);
+              assertThat_managedLibrariesContainsExactlyOne(libraries);
+            });
+  }
+
+  private void assertThat_managedLibrariesContainsExactlyOne(Set<CloudLibrary> managedLibraries) {
+    assertThat(managedLibraries.size()).isEqualTo(1);
+    CloudLibrary library = managedLibraries.iterator().next();
+    assertThat(library.getName()).isEqualTo(LIBRARY.name());
+  }
+
+  private void setModuleNameAndRoot(String root) {
+    moduleBuilder.setName("module");
+    moduleBuilder.setModuleFilePath(root + "/module.iml");
+    moduleBuilder.setContentEntryPath(root);
+  }
+
+  private Module createNewModule(MavenId id) {
+    moduleBuilder.setProjectId(id);
+
+    return ApplicationManager.getApplication()
+        .runWriteAction(
+            (Computable<Module>)
+                () -> {
+                  ModifiableModuleModel model =
+                      ModuleManager.getInstance(testFixture.getProject()).getModifiableModel();
+                  Module module = null;
+                  try {
+                    module = moduleBuilder.createModule(model);
+                  } catch (IOException
+                      | ModuleWithNameAlreadyExists
+                      | JDOMException
+                      | ConfigurationException e) {
+                    e.printStackTrace();
```

Comment:
Could we rather throw `AssertionError` here since this is completely unexpected in a unit test?

Reply:
nice catch; fixed